File: src-tauri/src/entity/project.rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "project")]
#[serde(rename_all = "camelCase")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i64,

    pub project_name: String,

    #[sea_orm(unique)]
    pub project_path: String,

    pub created_at: i64,

    pub last_opened_at: i64,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}

File: src/types/project.ts
export interface CreateProjectRequest {
  projectName: string;
  projectPath: string;
}

export interface Project {
  // 本机最近项目索引的 ID，不属于项目数据库中的业务数据。
  id: number;
  projectName: string;
  projectPath: string;
  createdAt: number;
  lastOpenedAt: number;
}

export interface QueryRecentOpenedProjectRequest {
  lastNum: number;
}
